### PR TITLE
Get lat-long from Google

### PR DESCRIPTION
**Pull request by longhotsummer**

### CODE REVIEW

File: mapit-za/views.py [hunk 1]
```diff
@@ -1,4 +1,84 @@
...
+        if address == "": return None
+
+        if not "south africa" in address.lower():
+            address = address + ", South Africa"
+
+        return self.resolve_address_google(address, **kwargs)
+
+
+def a2w(request):
+    address = request.GET['address']
```
Comment (longhotsummer) [inline mapit-za/views.py]: handle when there is no address, or when it's empty, or null. `request.GET.get('address')`

File: mapit-za/views.py [hunk 2]
```diff
@@ -1,4 +1,84 @@
...
+            return True
+        return False
+
+    def resolve_address_google(self, address, **kwargs):
+        import ipdb; ipdb.set_trace()
+        encoded_address = encode(address)
+        address = urllib.quote(encoded_address)
+
+        url = "https://maps.googleapis.com/maps/api/geocode/json?address=%s&sensor=false&region=za&key=%s" % (address, settings.GOOGLE_API_KEY)
+        response = urllib2.urlopen(url)
```
Comment (longhotsummer) [inline mapit-za/views.py]: lets use the requests library here. eg. see https://github.com/Code4SA/pmg-cms-2/blob/master/pmg/api_client.py#L63

**Comment by longhotsummer** [comment]
I've changed it so that if we find multiple locations, we just return all areas that match those locations. We now also support partial address matching by giving `partial=1` in the URL.

If we don't find anything, we just return an empty result set.

The MapIt API unfortunately doesn't make it easy to build on it, so there's a lot of cut and paste :(

**Comment by longhotsummer** [comment]
@xybrnet Let me know if you're happy and I'll merge

**Comment by xybrnet** [comment]
Cool, :+1:

**Pull request merged**